Mathlib/Data/List/Sort.lean

```diff
@@ -356,6 +368,27 @@ theorem sorted_insertionSort : ∀ l, Sorted r (insertionSort r l)
 
 end TotalAndTransitive
 
+/--
+If `c` is a sorted sublist of `l`, then `c` is still a sublist of `insertionSort r l`.
+-/
+theorem insertionSort_stable {l c : List α} (hr : c.Pairwise r) (hc : c <+ l) :
```

Comment:
This only needs the weaker `c <+~ l`, right?

Reply:
Presumably. The reason I chose `<+`, is because it's what is used for the statement of stability in https://github.com/leanprover/lean4/pull/5092.

Reply:
Probably, yes. I chose `<+` because that's what's used to state stability in https://github.com/leanprover/lean4/pull/5092. Would it be desirable to generalize to `<+~`?

Reply:
I have a proof of the more general version, using the following assumed lemmas:

```lean
theorem THM₁ [DecidableEq α] {l : List α} {p} (a : α) (h : l.Pairwise p) :
    (l.erase a).Pairwise p := by
  sorry

theorem THM₂ [DecidableEq α] {a : α} {as bs} (h : a :: as ~ bs) : as ~ bs.erase a := by
  sorry

theorem THM₃ {as bs} (a : α) (ha : as.Sorted r) (hs : as <+ bs) :
    orderedInsert r a as <+ orderedInsert r a bs := by
  sorry
```

Reply:
> If it's necessary, then perhaps it would be good to also keep the <+ version of the theorem.

Agreed.